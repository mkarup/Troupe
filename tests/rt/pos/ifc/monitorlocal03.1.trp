let val p = spawn (fn () => receive [ hn x => () ])
    val _ = let pini authority
                val _ =monitorlocal (p, `{secret}`)
            in ()
            end
    val _ = send (p, 42)
in
  rcvp(`{secret}`, [ hn x => x ]) (* Still need to receive at monitor level *)
end