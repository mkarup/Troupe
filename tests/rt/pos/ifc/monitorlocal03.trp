let val p = spawn (fn () =>
                      let val _ = if true raisedTo `{secret}` then
                                    receive [ hn x => () ]
                                  else ()
                      in () end)
    val monRef = monitorlocal (p, `{secret}`)
    val _ = send (p, 42)
in
  rcvp (`{secret}`, [ hn x => x ])
end