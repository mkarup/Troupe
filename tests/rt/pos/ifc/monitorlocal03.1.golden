2024-12-12T10:30:29.402Z [RTM] [32minfo[39m: Skipping network creation. Observe that all external IO operations will yield a runtime error.
>>> Main thread finished with value: ("DONE"@{}%{}, "4b8fe176-c8c3-47c7-89ea-42120acfce28"@{secret}%{}, e2e84c7f-1aac-40ff-b81e-a4f90024ea88@{}%{}@{secret}%{}, 0@{secret}%{})@{secret}%{}
